From dd3caa0dc3d7aa7aaabf1996d8b08ed4c9ab223f Mon Sep 17 00:00:00 2001
From: Jiri Olsa <jolsa@redhat.com>
Date: Tue, 19 Aug 2014 15:27:45 -0400
Subject: [tools] perf/trace: Change syscall summary duration order

Message-id: <1408462094-14194-297-git-send-email-jolsa@redhat.com>
Patchwork-id: 88268
O-Subject: [PATCH RHEL7.1 BZ1131394 296/325] perf trace: Change syscall summary duration order
Bugzilla: 1131394
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

From: Jiri Olsa <jolsa@kernel.org>

Bugzilla: 1131394
https://bugzilla.redhat.com/show_bug.cgi?id=1131394

upstream
========
commit 7f7a4138c66e857cc5bbf6e248a47379765349b5
Author: Pekka Enberg <penberg@kernel.org>
Date: Tue Nov 12 16:10:10 2013 +0200

description
===========
Switch duration order to minimum, average, maximum for the '--summary'
command line option because it's more natural to read.
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/tools/perf/builtin-trace.c b/tools/perf/builtin-trace.c
index 8d6ea8f..8990fbe 100644
--- a/tools/perf/builtin-trace.c
+++ b/tools/perf/builtin-trace.c
@@ -2093,7 +2093,7 @@ static size_t trace__fprintf_threads_header(FILE *fp)
  printed  = fprintf(fp, "\n _____________________________________________________________________________\n");
  printed += fprintf(fp, " __)    Summary of events    (__\n\n");
  printed += fprintf(fp, "              [ task - pid ]     [ events ] [ ratio ]  [ runtime ]\n");
- printed += fprintf(fp, "                                  syscall  count    min     max    avg  stddev\n");
+ printed += fprintf(fp, "                                  syscall  count    min     avg    max  stddev\n");
  printed += fprintf(fp, "                                                   msec    msec   msec     %%\n");
  printed += fprintf(fp, " _____________________________________________________________________________\n\n");
 
@@ -2129,8 +2129,8 @@ static size_t thread__dump_stats(struct thread_trace *ttrace,
    sc = &trace->syscalls.table[inode->i];
    printed += fprintf(fp, "%24s  %14s : ", "", sc->name);
    printed += fprintf(fp, "%5" PRIu64 "  %8.3f  %8.3f",
-        n, min, max);
-   printed += fprintf(fp, "  %8.3f  %6.2f\n", avg, pct);
+        n, min, avg);
+   printed += fprintf(fp, "  %8.3f  %6.2f\n", max, pct);
   }
 
   inode = intlist__next(inode);
-- 
1.7.1